tensorflow/python/eager/def_function_test.py

```diff
@@ -917,6 +917,50 @@ def f(x):
     self.assertLen(logs.output, 1)
     self.assertIn('Tracing is expensive', logs.output[0])
 
+  def test_type_annotation_basic(self):
+    def f(x: ops.Tensor[dtypes.Int32], y: ops.Tensor[dtypes.Float32]):
+      return x
+    signature = [tensor_spec.TensorSpec(None, dtypes.int32),
+                 tensor_spec.TensorSpec(None, dtypes.float32)]
+    def_function.function(f, input_signature=signature)
+
+  def test_mismatch_type_annotation(self):
```

Comment:
Nit: `test_mismatched` (attribute rather than verb).

Also, please name the tests with this pattern: `test_signature_validation_<test case>`